src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -106,13 +106,46 @@ dnnl::memory::data_type convert_data_type(cldnn::data_types dt) {
 }
 
 std::vector<std::pair<cldnn::format, dnnl::memory::format_tag>> format_map = {
-        { cldnn::format::bfyx, dnnl::memory::format_tag::nchw },
+        /// weights format for onednnn
+        { cldnn::format::oiyx,  dnnl::memory::format_tag::oihw },
+        { cldnn::format::ioyx,  dnnl::memory::format_tag::iohw },
+        { cldnn::format::yxio,  dnnl::memory::format_tag::hwio },
+        { cldnn::format::oizyx, dnnl::memory::format_tag::oidhw },
+        { cldnn::format::iozyx, dnnl::memory::format_tag::iodhw },
+        { cldnn::format::iyxo,  dnnl::memory::format_tag::ihwo },
+        { cldnn::format::oyxi,  dnnl::memory::format_tag::ohwi },
+        { cldnn::format::oyix,  dnnl::memory::format_tag::acbd },
+        { cldnn::format::oxiy,  dnnl::memory::format_tag::adbc },
+        { cldnn::format::goiyx,  dnnl::memory::format_tag::goihw },
+        { cldnn::format::gioyx,  dnnl::memory::format_tag::giohw },
+        { cldnn::format::gyxio,  dnnl::memory::format_tag::ghwio },
+        { cldnn::format::giozyx, dnnl::memory::format_tag::giodhw },
+        { cldnn::format::goizyx,  dnnl::memory::format_tag::goidhw },
+
+        { cldnn::format::os_iyx_osv16,  dnnl::memory::format_tag::Oihw16o },
+        { cldnn::format::gs_oiyx_gsv8,  dnnl::memory::format_tag::Goihw8g },
+        { cldnn::format::gs_oiyx_gsv16,  dnnl::memory::format_tag::Goihw16g },
+        { cldnn::format::gs_oiyx_gsv32,  dnnl::memory::format_tag::Goihw32g },
+        { cldnn::format::gs_oizyx_gsv16,  dnnl::memory::format_tag::Goidhw16g },
+        { cldnn::format::gs_oizyx_gsv32,  dnnl::memory::format_tag::Goidhw32g },
+        { cldnn::format::g_os_iyx_osv16,  dnnl::memory::format_tag::gOihw16o },
```

Comment:
why do we need such formats here?

Reply:
Actually, they don't need. these weights formats are maybe used in unit tests. So, I added a few more in advance.

Reply:
There are only a few formats to remove, so I'll leave them as is.